nengo_spinnaker/operators/value_source.py

```diff
@@ -144,12 +145,24 @@ def before_simulation(self, netlist, simulator, n_steps):
         outputs = []
         for conn in self.conns:
             output = []
+
+            # For each f(t) for the next set of simulations we calculate the
+            # output at the end of the connection.  To do this we first apply
+            # the pre-slice, then the function and then the post-slice.
             for v in values:
+                # If the value is not just a number we apply the pre-slice.
+                if not isinstance(v, numbers.Number):
+                    v = v[conn.pre_slice]
```

Comment:
I believe this will fail in this obscure case:

```python
    stim = nengo.Node(0)
    a = nengo.Ensemble(n_neurons=50, dimensions=2)
    nengo.Connection(stim[[0, 0]], a)
```